cmd/gitserver/server/list.go

```
@@ -48,50 +50,49 @@ func (s *Server) handleList(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) walkCloned(ctx context.Context, handle func(path string) error) error {
-	return filepath.Walk(s.ReposDir, func(path string, info os.FileInfo, err error) error {
-		if ctx.Err() != nil {
-			return ctx.Err()
-		}
+	return godirwalk.Walk(s.ReposDir, &godirwalk.Options{
+		Callback: func(path string, de *godirwalk.Dirent) error {
+			if ctx.Err() != nil {
+				return ctx.Err()
+			}
 
-		if s.ignorePath(path) {
-			if info.IsDir() {
-				return filepath.SkipDir
+			if s.ignorePath(path) {
+				if de.IsDir() {
+					return filepath.SkipDir
+				}
+				return nil
 			}
-			return nil
-		}
 
-		if err != nil {
-			return nil
-		}
```

you need to add an `ErrorCallback` so we ignore errors like we did with `filepath.Walk`. This error case was relatively easy to trigger due to:
- race conditions (some other command removes a file between listing and walking)
- permission errors (admins like to make our lives fun)

---

Ah! Thanks! If I understand the previous behavior correctly, we should probably return the `SkipNode` error action in the error callbacks: [see godoc here](https://godoc.org/github.com/karrick/godirwalk#ErrorAction). That skips the _node_ and this thus not `filepath.SkipDir`. That's the closest thing to "ignoring" I can find in godirwalk. What do you think?